content/courses/Ada_For_The_C_Embedded_Developer/chapters/05_SPARK.rst

```diff
@@ -159,100 +257,194 @@ profiles:
     void __gnat_last_chance_handler (char *source_location,
                                      int line);
 
-We'll go into the details of the :ada:`pragma Export` in a further section on
-language interfacing.
+We'll go into the details of the :ada:`pragma Export` in a further 
+section on language interfacing. For now, just know that the symbol 
+"__gnat_last_chance_handler" is what the run-time uses to branch 
+immediately to the last-chance handler. Pragma Export associates that 
+symbol with this replacement procedure so it will be invoked instead of 
+the default routine. As a result, the actual procedure name in Ada is 
+immaterial. 
+
+Here is an example implementation that simply blinks an LED
+forever on the target:
 
-Last, even without exception propagation, exceptions can still be handled
-locally, that is in the same scope as the one raising the exception. This
-allows the compiler to translate the exception handling through a simple
-:ada:`goto` statement. For example:
+.. code-block:: ada
+
+   procedure Last_Chance_Handler (Msg : System.Address; Line : Integer) is
+      pragma Unreferenced (Msg, Line);
+
+      Next_Release : Time := Clock;
+      Period       : constant Time_Span := Milliseconds (500);
+   begin
+      Initialize_LEDs;
+      All_LEDs_Off;
+
+      loop
+         Toggle (LCH_LED);
+         Next_Release := Next_Release + Period;
+         delay until Next_Release;
+      end loop;
+   end Last_Chance_Handler;
+
+The :ada:`LCH_LED` is a constant referencing the LED used by the
+last-chance handler, declared elsewhere. The infinite loop is necessary
+because a last-chance handler must never return to the caller (hence the
+term "last-chance"). The LED changes state every half-second.
+
+Unlike the approach in which there is only the last-chance handler 
+routine, the other approach also allows exception handlers but in a 
+specific idiomatic manner. In particular, whenever an exception is 
```

Comment:
Editorial suggestion: maybe we should also point to documentation that describes how to configure the compiler to use these two approaches.

Reply:
(GitHub won't let me reply to your question below...)

To create links, you have to use this format:
```
`LINK_TITLE <WWW_LINK>`_
```

Just as an example, the text below includes a link to the GPRbuild documentation:
```
Refer to the `GPRbuild User’s Guide <https://docs.adacore.com/gprbuild-docs/html/gprbuild_ug.html>`_ for more details.
```